ambry-store/src/main/java/com/github/ambry/store/StatsBasedCompactionPolicy.java

```diff
@@ -85,16 +101,70 @@ public CompactionDetails getCompactionDetails(long totalCapacity, long usedCapac
    * @return the {@link CostBenefitInfo} for the best candidate to compact. {@code null} if there isn't any.
    */
   private CostBenefitInfo getBestCandidateToCompact(NavigableMap<LogSegmentName, Long> validDataPerLogSegments,
-      long segmentCapacity, long segmentHeaderSize, long maxBlobSize) {
+      long segmentCapacity, long segmentHeaderSize, long maxBlobSize, BlobStoreStats blobStoreStats) {
     Map.Entry<LogSegmentName, Long> firstEntry = validDataPerLogSegments.firstEntry();
     Map.Entry<LogSegmentName, Long> lastEntry = validDataPerLogSegments.lastEntry();
     CostBenefitInfo bestCandidateToCompact = null;
+    boolean weightOnBenefit = false;
+
+    // weigh more on compaction benefit
+    // try to reclaim as many log segments as possible with reasonable cost.
+    if (storeConfig.storeStatsBasedMiddleRangeCompactionIntervalInMs != 0) {
```

Comment:
Can we separate out the two implementations (once where `weightOnBenefit` is `true`, and other where `weightOnBenefit` is `false`) into separate helper functions to make is easy to understand and make this function smaller.

Reply:
Added one function for middleRangeCompaction.